.github/workflows/publish-to-pypi.yml

```diff
@@ -0,0 +1,61 @@
+# Workflow for publishing archives to PyPI
+name: Publish to PyPI
+
+on:
+  # Runs for pull requests for testing purposes
+  pull_request:
+    branches:
+      - master
+  # Only runs for releases
+  release:
+    types:
+      - published
+
+jobs:
+  publish-pypi:
+    name: Publish to PyPI
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2.3.4
+      with:
+        # fecth all history so that versioneer works
+        fetch-depth: 0
+
+    - name: Set up Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.8
+
+    - name: Install dependencies
+      run: python -m pip install setuptools wheel
+
+    - name: Fix up version string
+      run: |
+        # Change the versioneer format to "pre" so that the commit hash isn't
+        # included (PyPI doesn't allow it). Can't do this permanently because
+        # we rely the hash to indicate to the tests that this is a local
+        # verison instead of a published version.
+        #
+        # The step is only necessary for testing purpose
+        sed --in-place "s/pep440/pep440-pre/g" setup.cfg
+
+    - name: Build source and wheel distributions
+      run: |
+        python setup.py sdist bdist_wheel
+        echo ""
+        echo "Generated files:"
+        ls -lh dist/
+
+    - name: Publish to Test PyPI
+      uses: pypa/gh-action-pypi-publish@v1.4.1
+      with:
+        password: ${{ secrets.test_pypi_password }}
+        repository_url: https://test.pypi.org/legacy/
```

Comment:
We may skip this step, as it's unnecessary to publish to test PyPI after we merge this PR into master.

Reply:
It's not necessary but it's nice to have this test. It avoids headaches when publishing a release and suddenly things fail because PyPI changed something.

Reply:
To have the test, I think I should keep the workaround `sed --in-place "s/pep440/pep440-pre/g" setup.cfg`, and trigger the workflow for pushes to master branch and tagged releases, right?

Reply:
👍

Reply:
The Publish to Test PyPI step will fail for the next few pushes to master (0.2.0.post0.dev52 onwards) until we release PyGMT v0.2.1 because it's trying to reupload with the same name. See e.g.
https://github.com/GenericMappingTools/pygmt/runs/1368511293?check_suite_focus=true#step:8:27

![image](https://user-images.githubusercontent.com/23487320/98450504-0c513800-21a2-11eb-8415-7dca3a9f2380.png)

A little annoying, but we can safely ignore it for now :slightly_smiling_face:

Reply:
Feel free to delete some releases.

Reply:
Hmm, I can't login to TestPyPI with my PyPI account, are they separate entities? Also seems like you're the only maintainer for PyGMT on TestPyPI.

Reply:
Yes they are separate. I will add you and leo as maintainers later.